packages/core/src/tracing/shimmer.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+// @ts-nocheck
+
+'use strict';
+
+const shimmer = require('shimmer');
+let logger;
+logger = require('../logger').getLogger('tracing/shimmer', newLogger => {
+  logger = newLogger;
+});
+
+exports.unwrap = shimmer.unwrap;
+
+/**
+ * This wrapper prevents that the application of the customer dies.
+ *
+ * We won't catch errors which happened e.g. when the result of the original fn comes back.
+ * This is fine, because we want to primarly catch errors which happened before the library call.
+ */
+exports.wrap = (origObject, origMethod, instrumentationWrapperMethod) => {
+  shimmer.wrap(origObject, origMethod, function shimmerWrapper(originalFunction) {
+    return function () {
```

Comment:
_nitpick:_ Do we want this to be an anonymous function or should we make it a named function as well?

Reply:
Yes we want named fn's everywhere <3 Good catch!